File: Kotlin_sample/app/src/main/java/com/example/kotlin_sample/code_generator/AccessCodeGenerator.kt
package com.example.kotlin_sample.code_generator

import java.security.MessageDigest
import java.text.SimpleDateFormat
import java.util.Base64
import java.util.Locale
import java.util.TimeZone
import java.util.Date


fun generateAccessCode(appID: String, utcTime: Date): String {
//  Generates the Access Code from the app id and the current time.
//   Used when trying to access the receiver API or any API that requires it.
  val lowercaseID = appID.lowercase(Locale.getDefault())

  val iso8601Format = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.US).apply {
    timeZone = TimeZone.getTimeZone("UTC")
  }
  val iso8601Time = iso8601Format.format(utcTime)

  val plaintextAccessCode = lowercaseID + iso8601Time
  val utf8Bytes = plaintextAccessCode.toByteArray(Charsets.UTF_8)
  val hashedBytes = MessageDigest.getInstance("SHA-256").digest(utf8Bytes)
  val base64String = Base64.getEncoder().encodeToString(hashedBytes)

  return base64String
}

File: Maui-sample/AccessCode/AccessCodeGenerator.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Security.Cryptography;

namespace Maui_sample.AccessCode
{
    class AccessCodeGenerator
    {
    private string _accessCode;

    public string ID { get; private set; }
    public DateTime UtcTime { get; private set; }
    public string AccessCode => _accessCode ??= GenerateAccessCode();

    public AccessCodeGenerator(string id, DateTime utcTime)
    {
      if (utcTime.Kind != DateTimeKind.Utc)
      {
        throw new ArgumentException("utcTime must be of DateTimeKind.Utc kind", nameof(utcTime));
      }
      ID = id;
      UtcTime = utcTime;
    }

    public string GenerateAccessCode()
    {
      return GenerateAccessCode(ID, UtcTime);
    }

    public static string GenerateAccessCode(string appID, DateTime utcTime)
    {
      // Generates access code when trying to acccess receiver API or any API that requires it.
      string lowercaseID = appID.ToLowerInvariant();
      // Format utcTime as an ISO8601 compliant string, like this: 2024-02-22T18:00:00Z
      string iso8601Time = utcTime.ToString("yyyy-MM-dd'T'HH:mm:ssK", CultureInfo.InvariantCulture);
      string plaintextAccessCode = lowercaseID + iso8601Time;
      byte[] utf8Bytes = Encoding.UTF8.GetBytes(plaintextAccessCode);
      byte[] hashedBytes = SHA256.HashData(utf8Bytes);
      string base64String = Convert.ToBase64String(hashedBytes);
      return base64String;
    }
  }
}
